src/composables/notifications.ts

```diff
@@ -0,0 +1,155 @@
+  );
+
+  const filteredNotifications = computed<INotification[]>(
+    () => [...superheroNotifications.value, ...walletNotifications.value]
+      .filter(
+        ({ status, type }) => (
+          status === NOTIFICATION_STATUS_READ
+                  || notificationSettingsCheckedTypes.value.includes(type)
+        ),
+      )
+      .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
+  );
+
+  const notificationsToShow = computed<INotification[]>(
+    () => filteredNotifications.value.slice(0, offset.value * CHUNK_SIZE),
+  );
+
+  async function getSuperheroNotifications() {
+    const chainNames = useState('chainNames');
+
+    superheroNotifications.value = (
+      await getAllNotifications(sdk.value, activeNetwork.value, account.value.address)
+    ).map(normalizeNotification(chainNames.value));
+  }
+
+  function addWalletNotification(payload: Partial<INotification>) {
+    walletNotifications.value.push({
+      ...payload,
+      type: NOTIFICATION_TYPE_WALLET,
+      status: NOTIFICATION_STATUS_CREATED,
+      createdAt: new Date().toISOString(),
+    });
+  }
+
+  function setWalletNotificationsStatus(createdAt: string, status: NotificationStatus) {
+    const currentNotification = walletNotifications.value.find(
+      (notification) => notification.createdAt === createdAt,
+    );
+    if (currentNotification) {
+      currentNotification.status = status;
+    }
+  }
+
+  function loadMore() {
+    canLoadMore.value = false;
+    offset.value += 1;
+    if (notificationsToShow.value.length < filteredNotifications.value.length) {
+      canLoadMore.value = true;
+    }
+  }
+
+  function handleSetPollingInterval() {
+    pollingInterval.value = executeAndSetInterval(() => getSuperheroNotifications(), 30000);
+  }
+
+  function handleClearPollingInterval() {
+    if (pollingInterval.value) {
+      clearInterval(pollingInterval.value);
+    }
+  }
+
+  async function handleMarkAsReadSuperhero() {
+    const status = NOTIFICATION_STATUS_READ;
+
+    const unreadNotificationsIds = superheroNotifications.value
+      .filter((notification) => notification.status !== status)
+      .map((notification) => notification.id!);
+
+    superheroNotifications.value = superheroNotifications.value.map((notification) => ({
+      ...notification,
+      status,
+    }));
+
+    await modifyNotifications(
+      sdk.value,
+      activeNetwork.value,
+      account.value.address,
+      unreadNotificationsIds,
+      status,
+    );
+    handleClearPollingInterval();
+    handleSetPollingInterval();
+  }
+
+  function handleMarkAsReadWallet() {
+    walletNotifications.value.forEach((notification) => {
+      if (notification.type === NOTIFICATION_TYPE_WALLET) {
+        setWalletNotificationsStatus(
+          notification.createdAt,
+          NOTIFICATION_STATUS_READ,
+        );
+      }
+    });
+  }
+
+  function handleMarkAsRead() {
```

Comment:
And again I suggest to remove the 'handle' word :)
Maybe the name would make more sense as `markAllAsRead`?